src/libs/NextStepUtils.ts

```diff
@@ -78,7 +79,16 @@ function buildNextStep(report: OnyxEntry<Report>, predictedNextStatus: ValueOf<t
     const autoReportingFrequency = PolicyUtils.getCorrectedAutoReportingFrequency(policy);
     const submitToAccountID = PolicyUtils.getSubmitToAccountID(policy, ownerAccountID);
     const ownerDisplayName = ReportUtils.getDisplayNameForParticipant(ownerAccountID);
-    const managerDisplayName = ReportUtils.getDisplayNameForParticipant(submitToAccountID);
+    let nextApproverDisplayName = ReportUtils.getDisplayNameForParticipant(submitToAccountID);
+
+    const approvalChain = PolicyUtils.getApprovalChain(policy, ownerAccountID, report?.total ?? 0);
+    if (approvalChain.length > 0) {
+        const nextApproverEmail = approvalChain.length === 1 ? approvalChain[0] : approvalChain[approvalChain.indexOf(currentUserEmail) + 1];
```

Comment:
Wondering if this `+1` on the index could ever be a problem?

Reply:
it wont be. if index overflows, there are no more approvers and nextApprover is meaningless in that context

Reply:
Maybe here is where we'd need a "You" if we're currently waiting on the logged in user to approve the report?

Reply:
@Beamanator no. The next steps are not dynamic. Everyone sees them as the same.
Ultimately, these are **optimistic** next steps. The response from backend will still say "Waiting for UserX to approver..."

Reply:
... why can't we optimistically know if it's "you" or not?

Reply:
Chain: Alex -> Rushat -> Manan

@Beamanator the optimistic steps are generated at the time of approving a report.

When Rushat approves Alex's expenses the optimistic message will be "Waiting for Manan to approve".
When Manan is logged in, we are not generating the next steps again. We will just show the backend response for next steps.

Reply:
oh i think i get it... whennn it's "your" turn to approve, the backend will return the "you" 👍 (via pusher or openReport or something like that). Whenever you approve, you won't ever "optimistically" become the approver.

... Unless you submit to yourself 👀

Reply:
>  Unless you submit to yourself 👀

that's a good catch. The backend does not handle it either

<img width="662" alt="image" src="https://github.com/user-attachments/assets/d1b0b4b2-b8da-466b-927c-df15010c0715">

Reply:
> oh i think i get it... whennn it's "your" turn to approve, the backend will return the "you" 👍

I think it should but it does not.

> Whenever you approve, you won't ever "optimistically" become the approver

that's correct. with an exception that you [mentioned](https://github.com/Expensify/App/pull/44940#discussion_r1706324267)